src/python/nimbusml/pipeline.py

```diff
@@ -629,108 +629,164 @@ def _update_graph_nodes_for_learner(
-                    if label_column is None:
-                        label_column = Role.Label
-                    last_node.label_column_name = label_column
+                learner_features = [learner.feature_column_name]
+        elif strategy_iosklearn in ("previous", "accumulate"):
+            if hasattr(
+                    learner,
+                    'feature') and learner.feature is not None:
+                if isinstance(learner.feature, list):
+                    learner_features = learner.feature
                 else:
-                    label_column = _extract_label_column(
-                        last_node, DataSchema.read_schema(y))
-                    if label_column is None:
-                        label_column = Role.Label
-                    last_node.label_column_name = label_column
-            else:
-                last_node.label_column_name = None
-                label_column = None
-
-            if weight_column is not None or last_node._use_role(Role.Weight):
-                if getattr(last_node, 'example_weight_column_name', None):
-                    weight_column = last_node.example_weight_column_name
-                elif weight_column:
-                    last_node.example_weight_column_name = weight_column
+                    learner_features = [learner.feature]
+                learner.feature_column_name = 'Features'
+            elif isinstance(columns_out, list):
+                learner_features = columns_out
+                learner.feature_column_name = 'Features'
+            elif columns_out is None:
+                learner_features = ['Features']
+                learner.feature_column_name = 'Features'
             else:
-                last_node.example_weight_column_name = None
-                weight_column = None
-
-            if (hasattr(last_node, 'row_group_column_name_')
-                    and last_node.row_group_column_name_ is not None):
-                group_id_column = last_node.row_group_column_name_
-            elif (hasattr(last_node, 'row_group_column_name') and
-                  last_node.row_group_column_name is not None):
-                group_id_column = last_node.row_group_column_name
+                learner_features = [columns_out]
+                learner.feature_column_name = 'Features'
+        else:
+            raise NotImplementedError(
+                "Strategy '{0}' to handle unspecified inputs is not "
+                "implemented".format(strategy_iosklearn))
+
+        if label_column is not None or learner._use_role(Role.Label):
+            if getattr(learner, 'label_column_name_', None):
+                label_column = learner.label_column_name_
+            elif getattr(learner, 'label_column_name', None):
+                label_column = learner.label_column_name
+            elif label_column:
+                learner.label_column_name = label_column
+            elif y is None:
+                if label_column is None:
```

Comment:
this is crazy :)

Reply:
This is the same code that was there before my change. The only modification I made was to unindent this code by 4 spaces to remove an unnecessary `if` statement.

I will look in to what can be done to clean this up. Though, I might put that in a different pull request.

---
In reply to: [361725035](https://github.com/microsoft/NimbusML/pull/390#discussion_r361725035) [](ancestors = 361725035)